python/src/com/jetbrains/python/codeInsight/controlflow/PyControlFlowBuilder.java

```diff
@@ -738,7 +738,7 @@ public void visitPyAssertStatement(final PyAssertStatement node) {
     super.visitPyAssertStatement(node);
     final PyExpression[] args = node.getArguments();
     // assert False
-    if (args.length >= 1 && PyConstantExpressionEvaluator.evaluate(args[0]) == Boolean.FALSE) {
+    if (args.length >= 1 && PyConstantExpressionEvaluator.evaluateBoolean(args[0], true)) {
```

Comment:
I suppose you forgot about "!" before PyConstantExpressionEvaluator.evaluateBoolean(args[0], true) :)